### ISSUE TITLE
develop dipolesdb.py

### ISSUE THREAD

**Issue by rreho** [yambo-code/yambopy#24]
I have some concerns regarding the calculation of absorption in `dipolesdb.py`. The function `ip_eps2` is calculating absorption and not the dielectric function and it works only if we use `dip_type='iR'`, right?

I would like to calculate absorption starting from the dipoles database of Yambo, either in length or velocity gauge but the correct formulas that I would have to use starting from the Yambo `ndb.dipoles` are not clear to me.

I think that in the velocity gauge I would have to use
$`
\epsilon_{\alpha, \alpha}(\omega)=1+\frac{16 \pi}{\Omega} \sum_{c, v} \sum_{\mathbf{k}} \frac{1}{E_{c \mathbf{k}}-E_{v \mathbf{k}}} \frac{\left|\left\langle v \mathbf{k}\left|\mathbf{p}_\alpha+i\left[V^{\mathrm{NL}}, \mathbf{r}_\alpha\right]\right| c \mathbf{k}\right\rangle\right|^2}{\left(E_{c \mathbf{k}}-E_{v \mathbf{k}}\right)^2-(\omega+i \gamma)^2}
`$
while I am not sure about the same formula in the length gauge. If I understood it correctly, 'ip_eps2' is  using this formula for absorption
$`
Abs(\omega)=\sum_{c, v}\frac{\left|\left\langle v \mathbf{k}\left|\mathbf{r}_\alpha\right| c \mathbf{k}\right\rangle\right|^2}{\left(E_{c \mathbf{k}}-E_{v \mathbf{k}}\right)^2-(\omega+i \gamma)^2}
`$
without considering volume factors etc. Do you think this is correct?

I propose to add the calculation of the dielectric response function and then extract from its imaginary part absorption. In this way, we can also properly take into account of the geometrical factor needed for the 0D 1D, and 2D cases and calculate also the absorptance, absorbance, etc..
What I need is a formula that, starting from the `DIP_iR` and `DIP_ip` in Yambo gives me $`\epsilon_{ij}`$. 
Do you have any suggestions/comments?

**Comment by palful** [comment]
[This yambo development](https://github.com/yambo-code/yambo-devel/pull/765#pullrequestreview-1862283218) is relevant to this discussion as it concerns the calculation of `alpha`. Yambopy should mirror it in `excitonDB` and, if applicable, in `dipolesDB`.

The goal is to have the yambopy plots of `eps_1(w)` and `eps_2(w) / alpha(w)` (i.e. absorption) coincide with the yambo `o-*` files in both 3D and 2D.

**Comment by palful** [comment]
Also, the new version of `dipolesDB` finally can read spin-polarized dipoles.

**Comment by palful** [comment]
Update: the missing factor in the IPA of the 3D case is fixed. It was due the kpoint weights being rescaled by Nk_IBZ instead of Nk_BZ. I have also changed the way the dipoles are projected along the field direction.

As for the 2D case, we still have a constant factor missing, so, even if I added the case of alpha_2D, so far it doesn't matter much. I am closing this issue even if the 2D case is not perfectly solved.
